state-chain/pallets/cf-ingress-egress/src/lib.rs

```diff
@@ -375,6 +380,15 @@ pub mod pallet {
 		pub boost_fee: BasisPoints,
 	}
 
+	#[derive(
+		CloneNoBound, RuntimeDebugNoBound, PartialEqNoBound, EqNoBound, Encode, Decode, TypeInfo,
+	)]
+	#[scale_info(skip_type_params(T, I))]
+	pub enum DepositFailedDetails<T: Config<I>, I: 'static> {
+		DepositChannel { deposit_witness: DepositWitness<T::TargetChain> },
+		Vault { vault_witness: Box<VaultDepositWitness<T, I>> },
```

Comment:
I'm assuming we put VaultDepositWitness in a Box to reduce the size of `DepositFailedDetails` on the stack? If so, why not also box `DepositWitness`? I know `VaultDepositWitness` is larger, but the other struct isn't particularly small either. Just curious if we have a rule of thumb for things like this @dandanlen. Or alternatively, are we enforcing max size for the error enum like we do for runtime calls?

Reply:
Fwiw the reason I used a Box was just clippy's advise due to the usual "large size difference between variants".